yaml_list.yaml

```diff
@@ -1,7 +1,14 @@
 yaml_test_list:
```

Comment:
The code to loop through the items in this file:
```python
for test in config['yaml_test_list']:
```
has been removed, so might as well remove the file too.

Similarly, the `nose-testconfig` requirements in `setup.py`, and the `tc-file` and `tc-format` options in `setup.cfg` won't be needed anymore.